src/validators/mod.rs

```diff
@@ -50,7 +51,17 @@ impl SchemaValidator {
             }
         };
         let slots = slots_builder.into_slots()?;
-        Ok(Self { validator, slots })
+        Ok(Self {
+            validator,
+            slots,
+            schema: schema.into_py(py),
+        })
+    }
+
+    fn __reduce__(&self, py: Python) -> PyResult<PyObject> {
+        let args = (self.schema.as_ref(py),);
+        let cls = Py::new(py, self.to_owned())?.getattr(py, "__class__")?;
+        Ok((cls, args).into_py(py))
```

Comment:
these changes are just slightly more succinct while I tried to avoid the to_owned/clone above, still probably worth it.